src/GitVersionCore/VersionCalculation/MainlineVersionCalculator.cs

```diff
@@ -72,7 +71,7 @@ public SemanticVersion FindMainlineModeVersion(BaseVersion baseVersion)
                 mainlineVersion.BuildMetaData = CreateVersionBuildMetaData(mergeBase);
 
                 // branches other than master always get a bump for the act of branching
-                if ((!context.CurrentBranch.IsSameBranch(mainline)) && (string.IsNullOrEmpty(context.Configuration.NextVersion)))
+                if ((!context.CurrentBranch.Equals(mainline)) && (string.IsNullOrEmpty(context.Configuration.NextVersion)))
```

Comment:
Can we use `!=` instead of `!Equals()` here? I find explicit calls to `.Equals(object)` buggy since it accepts `object` and thus doesn't provide compile-time checks for comparing different types (which would always return `false`).

```suggestion
                if ((context.CurrentBranch != mainline) && (string.IsNullOrEmpty(context.Configuration.NextVersion)))
```

Reply:
I think Resharper was complaining about that saying about reference comparison and was suggesting to use Equals instead.

Reply:
I tried to replace `Equal` with `==` and the tests are failing, and I guess that is because the `==` operator is implemented on the `Branch` class but we compare the `IBranch`, so I will keep using Equals for now